gensim/sklearn_api/basemodel.py

```diff
@@ -11,7 +11,7 @@
 from abc import ABCMeta, abstractmethod
 
 
-class BaseSklearnWrapper(object):
+class BaseTransformer(object):
```

Comment:
In my head I expect all sklearn models to have `partial_fit`, `fit`, `transform` and be used in a pipeline. We constantly get questions on the mailing list about `how to update model X` for almost every sklearn algo. Raising a `NotImplemented ` with an explanation for word2vec and doc2vec `use gensim api update function but it's not recommended as it hasn't been researched.`  And throw an explicitly that RandomProj can't do it because of algo limitations.